### PR TITLE
Cleanup and bug fixes

### PR DESCRIPTION
**Pull request by EdsterG**
When resuming training at iteration N, the random state must also be restored.

### CODE REVIEW

File: src/proto/gps.proto [hunk 1]
```diff
@@ -23,7 +23,8 @@ enum SampleType {
   IMAGE_FEAT = 16;
   END_EFFECTOR_POINTS_NO_TARGET = 17;
   END_EFFECTOR_POINT_VELOCITIES_NO_TARGET = 18;
-  TOTAL_DATA_TYPES = 19;
+  INDICATOR = 19;
```
Comment (cbfinn) [inline src/proto/gps.proto]: what is this for?

File: python/gps/algorithm/policy_opt/config.py [hunk 2]
```diff
@@ -38,12 +38,9 @@
 POLICY_OPT_CAFFE.update(GENERIC_CONFIG)
 
 
-checkpoint_path = os.path.abspath(os.path.join(os.path.dirname(__file__),
```
Comment (cbfinn) [inline python/gps/algorithm/policy_opt/config.py:41]: this shouldn't be removed...

**Review by cbfinn — CHANGES_REQUESTED** [review]
The change with the random state looks good. The other changes shouldn't be added.

File: src/proto/gps.proto (hunk 1 above)
Comment (EdsterG) [reply src/proto/gps.proto]: Indicator sensor for "object grasped". It's used when training in sim.

File: python/gps/algorithm/policy_opt/config.py (hunk 2 above)
Comment (EdsterG) [reply python/gps/algorithm/policy_opt/config.py:41]: it's not used anywhere, why do we need this line?

File: src/proto/gps.proto (hunk 1 above)
Comment (cbfinn) [reply src/proto/gps.proto]: I don't think we should add it to the public repo unless it is used somewhere.

File: python/gps/algorithm/policy_opt/config.py (hunk 2 above)
Comment (cbfinn) [reply python/gps/algorithm/policy_opt/config.py:41]: It's used right here: https://github.com/cbfinn/gps/blob/master/python/gps/algorithm/policy_opt/policy_opt_tf.py#L34

Comment (EdsterG) [reply python/gps/algorithm/policy_opt/config.py:41]: self.checkpoint_file isn't referenced anywhere. A few commits ago it was used with self.save.save but now the weights are saved in a manner that doesn't require a checkpoint_file. Do you plan on returning that functionality?

Comment (cbfinn) [reply python/gps/algorithm/policy_opt/config.py:41]: Ok, that's fine. Please remove self.checkpoint_file then too!

Comment (EdsterG) [reply python/gps/algorithm/policy_opt/config.py:41]: An alternative option is to make PolicyOptTf similar to PolicyOptCaffe.
Change 'checkpoint_file' to 'checkpoint_file_prefix' or 'weights_file_prefix' and then use it in getstate/setstate just like it's used in PolicyOptCaffe?

Comment (cbfinn) [reply python/gps/algorithm/policy_opt/config.py:41]: That sounds good to me.

Comment (cbfinn) [reply python/gps/algorithm/policy_opt/config.py:41]: Actually, nevermind. The way that it is coded right now is best, so you never run into file conflicts and the weights are stored with the policy. Caffe should be changed to do it how tf does it, if possible.

**Comment by EdsterG** [comment]
I've updated gps.proto

**Comment by cbfinn** [comment]
Can you also delete this line?
https://github.com/cbfinn/gps/blob/master/python/gps/algorithm/policy_opt/policy_opt_tf.py#L34

Other than that, LGTM!

**Comment by EdsterG** [comment]
Done!

**Comment by wmontgomery4** [comment]
Don't think you meant to close it ;) Merging, I've been following the PR and it LGTM as well.

**Pull request merged**